### DISCUSSION TITLE
Table with checkboxes

### DISCUSSION THREAD

**Discussion by fousasg** [h2oai/wave#1943] [Q&A]
Hello, 

I am trying to create an app with a table with checkboxes in its cells (for many columns) so that the user can make selections.

Like the below:
![Screenshot from 2023-04-25 13-00-31](https://user-images.githubusercontent.com/20576978/234296530-43b72cf7-bd5c-450f-8126-0c836eee9a0a.png)

Then I want to parse that table and use it in the next step.

Is there a way to do that?

**Comment by mturoci** [comment]
Hi @fousasg. Wave supports a single-column interactive checkbox only - https://wave.h2o.ai/docs/widgets/form/table/#checkbox-visibility. Can you elaborate on how the multi-col selection should work? Currently, it's very simple - you either check or not check the row which makes up the whole selection.

**Comment by fousasg** [comment]
Ideally, I would like to give to the app user a table, like the one presented above, that would be able to "check" some of the columns and then be able to get back those selections.  

An idea could be to allow a table to have a column that could have a **"cell type"** of a **checkbox**. The same way that columns can be **tags** or **processes**.

**Reply by mturoci** [reply]
> that would be able to "check" some of the columns and then be able to get back those selections.

hm.. can you send me a real-world example of this?

> The same way that columns can be tags or processes.

Table currently submits `q.args.<table-name> = [clicked_row_id1, clicked_row_id2...]`. How would the checkboxes align with that?

**Comment by fousasg** [comment]
A real-world example could be that we have a list of variables, and we want to check the ones that should be included in a model-building process and, at the same time, pass modelling constraints (for example, monotonicity) that will be used as modelling settings, like below:
| Variable      | Select | Positive monotonicity | Negative monotonicity |
| ----------- | ----------- | ----------- | ----------- |
| Age      | <ul><li>- [x] </li></ul>        | <ul><li>- [x] </li></ul>        |<ul><li>- [ ] </li></ul>|
| Income   | <ul><li>- [x] </li></ul>        | <ul><li>- [x] </li></ul>        |<ul><li>- [ ] </li></ul>       |
| Debt   | <ul><li>- [x] </li></ul>       | <ul><li>- [ ] </li></ul>      | <ul><li>- [x] </li></ul>       |
| Marital status   | <ul><li>- [x] </li></ul>       | <ul><li>- [ ] </li></ul>      | <ul><li>- [ ] </li></ul>      |
| Nationality   | <ul><li>- [ ] </li></ul>      | <ul><li>- [ ] </li></ul>     | <ul><li>- [ ] </li></ul>    |

**Reply by mturoci** [reply]
I mean if you could link a similar component / UX - flow used in other web apps. Just asking since I haven't seen it before (or maybe don't realize).

**Reply by fousasg** [reply]
Oh I see, let me have a look.

**Comment by fousasg** [comment]
I am not sure if the below help.

https://community.plotly.com/t/add-checkbox-in-a-dash-datatable/52600

https://www.pythonguis.com/faq/abstract-table-model-question/

**Reply by mturoci** [reply]
By real-world I mean whether such things are present in popular web apps like Google or maybe other AI platforms. I am asking because I haven't seen such UX pattern (or don't realize it) yet, which implies there might be a better alternative for doing so.
